src/mgmt/xmgmt-fmgr-drv.c

```diff
@@ -115,10 +116,20 @@ static int xmgmt_pr_write_complete(struct fpga_manager *mgr,
 		return -EINVAL;
 	}
 
+	axigate_leaf = xocl_subdev_get_leaf_by_id((struct platform_device *)obj->pdev, XOCL_SUBDEV_AXIGATE,
+						  PLATFORM_DEVID_NONE);
```

Comment:
We have two instances of AXI-gate leaf drivers. Using PLATFORM_DEVID_NONE will just get you the first one (this would be fine for most of code since we normally have only one instance for one type of leaf driver), which may or may not be the the one you want. I think we want to use the event callback and provide a filter function for framework to return the correct instance. @houlz0507 , what should we be looking at to tell if this AXI-gate is what we want?

Reply:
It should used the resource name to determine the gate

Reply:
There is an inline function in xocl-axigate.h you may use
static inline bool xocl_axigate_match_epname(enum xocl_subdev_id id,
        struct platform_device *pdev, void *arg)

Reply:
Changed to `axigate_leaf = xocl_subdev_get_leaf((struct platform_device *)obj->pdev, xocl_axigate_match_epname, NODE_GATE_ULP);`